### ISSUE TITLE
랜덤포레스트

### ISSUE THREAD

**Issue by eungjukim** [eungjukim/SMU_business_bigdata_analysis#9]
@pyj5485 님이 의사결정나무는 사용할 수 없느냐는 의견을 주셨습니다. 처음에는 의사결정나무를 사용할 수 없다고 판단했는데, 로지스틱 회귀분석을 진행하면서 생각해보니 의사결정나무를 통해서도 고객의 만족도를 예측할 수 있다고 생각했습니다. 그래서 의사결정나무의 한 종류인 랜덤포레스트 기법을 적용하고자 합니다. 기본적인 변수들은 이전의 로지스틱 회귀분석과 동일합니다.

**Comment by eungjukim** [comment]
랜덤포레스트 코드입니다.
```python
#random forest
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

rf = RandomForestClassifier(n_estimators=100, oob_score=True, random_state=123456)
rf.fit(X_train, y_train)
predicted = rf.predict(X_test)
accuracy = accuracy_score(y_test, predicted)

print(f'Out-of-bag score estimate: {rf.oob_score_:.3}')
print(f'Mean accuracy score: {accuracy:.3}')

cm = pd.DataFrame(confusion_matrix(y_test, predicted))
sns.heatmap(cm, annot=True)
```
#참고한 사이트 : https://partrita.github.io/posts/random-forest-python/

**Comment by eungjukim** [comment]
랜덤포레스트 결과입니다.
Out-of-bag score estimate: 0.957
Mean accuracy score: 0.958
<matplotlib.axes._subplots.AxesSubplot at 0x24a095ba160>

**Comment by eungjukim** [comment]
랜덤포레스트 모델의 예측력이 0.95 정도로 굉장히 높았습니다. 이전의 로지스틱 회귀분석 모델의 성능(0.86)도 굉장히 높다고 생각했는데, 더욱 높은 수치가 나왔네요. 만족스럽습니다.

**Comment by pyj5485** [comment]
예측력이 높아졌네요! 수고하셨어요~

**Comment by eungjukim** [comment]
![image](https://user-images.githubusercontent.com/42957465/50130397-c4308000-02c0-11e9-8058-f896563136ea.png)
